src/labone/core/session.py

```diff
@@ -497,14 +484,31 @@ async def _multi_value_postprocessing(
         response = (await future).result
         try:
             return [
-                AnnotatedValue.from_capnp(unwrap(raw_result)) for raw_result in response
+                AnnotatedValue.from_capnp(
+                    t.cast(hpk_schema.AnnotatedValue, unwrap(raw_result)),
+                )
+                for raw_result in response
             ]
         except RuntimeError as e:
             msg = f"Error while setting {path}."
             raise errors.LabOneCoreError(msg) from e
 
+    @t.overload
+    def set(self, value: AnnotatedValue) -> t.Awaitable[AnnotatedValue]: ...
+
+    @t.overload
+    def set(
+        self,
+        value: Value,
+        path: LabOneNodePath,
+    ) -> t.Awaitable[AnnotatedValue]: ...
+
     @translate_comms_error
-    def set(self, value: AnnotatedValue) -> t.Awaitable[AnnotatedValue]:
+    def set(
+        self,
+        value: AnnotatedValue | Value,
+        path: LabOneNodePath | None = None,
```

Comment:
Should we raise an Exception if `value` is `AnnotatedValue` and `path` is given.

Or another option is to document that whenever `AnnotatedValue` is used, `path` value is ignored.

Reply:
Isn't this documented by the overloaded function declarations?